### ISSUE TITLE
when i update a plugin by ftp, the package.json not be update.

### ISSUE THREAD

**Issue by zynfly** [cedaro/satispress#131]
when i update a plugin by ftp, the package.json not be update, but  the page settings.php?page=satispress#satispress-packages  is ok.

**Comment by cfsegovia** [comment]
Hi, I ran in the same problem, the trick is got to admin of packages and manually download, after that the packages are updated in the JSON, maybe the Team of satispress can  put a hook to generate automatic the zip file and update the JSON file or cronjob for that, some ideas

**Comment by tjedrak** [comment]
I have the same issue, whenever I upload newest version by FTP or by Upload plugin (from Plugin panel) -  .json file doesn't get any updates.

**Comment by JanThiel** [comment]
Same here. We deploy and update custom plugins using GIT or deploy scripts. Both do not create the necessary package files and thus using composer with these new versions fail.

Using WP-CLI or a WP-CRON to run the creation process would be a great way to automate this. Manually clicking a link doesn't really scale ;-)

Thanks and best regards,
Jan

**Comment by iljamm** [comment]
same here, I uploaded an updated version of a plugin but it would not add that new version to packages.json file. as per @cfsegovia commant, downloading a new version of a plugin does force update of packages.json file.

**Comment by danielbachhuber** [comment]
Same issue for me 😄 If I replace the plugin via the "Upload Plugin" UX, the `package.json` isn't refreshed.

**Comment by bradyvercher** [comment]
I assumed if anyone was updating packages manually, they would generate the artifact manually since there isn't really a hook to monitor the filesystem. WP-CRON is an option, but it would still likely cause some confusion as well since it might take awhile for the updated release to become available.

The download route attempts to create artifacts on demand if they don't exist. I took the same approach when viewing `packages.json` so that the installed release could be cached automatically if it doesn't exist. That should cover all the use cases raised here (FTP, git, deploy scripts, upload through the admin UI).

I also added a routine when upgrading through the admin UI to cache the latest release for redundancy and to make sure it exists in the event any future functionality expects it to be there.
